### PR TITLE
[Web PubSub] Update to API version 2022-08-01-preview

### PR DESCRIPTION
**Pull request by Y-Sindo**
## Description

Updated to API version 2022-08-01-preview
* Added support for custom domain. Added new cmdlets New-AzWebPubSubCustomCertificate, Get-AzWebPubSubCustomCertificate, Remove-AzWebPubSubCustomCertificate, New-AzWebPubSubCustomDomain, Get-AzWebPubSubCustomDomain, Remove-AzWebPubSubCustomDomain.
* Added support for event listeners in hub settings. Added new cmdlets New-AzWebPubSubEventHubEndpointObject, New-AzWebPubSubEventNameFilterObject.

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by BethanyZhou** [comment]
@Nickcandy please help review

**Comment by Nickcandy** [comment]
@BethanyZhou Hi Beisi, it looks good to me

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

File: src/SignalR/SignalR.Autorest/custom/scripts/New-AzWebPubSubEventHubEndpointObject.ps1 [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+Create an in-memory object for EventHubEndpoint.
+.Description
+Create an in-memory object for EventHubEndpoint.
+
+.Outputs
+Microsoft.Azure.PowerShell.Cmdlets.WebPubSub.Models.Api20220801Preview.EventHubEndpoint
+.Link
+https://learn.microsoft.com/powershell/module/az./new-AzWebPubSubEventHubEndpointObject
+#>
+function New-AzWebPubSubEventHubEndpointObject
```
[the remark this answers is not in the document]
Comment (Y-Sindo) [reply src/SignalR/SignalR.Autorest/custom/scripts/New-AzWebPubSubEventHubEndpointObject.ps1:28]: The cmdlets generated by autorest is `New-AzEventHubEndpointObject` (without *WebPubSub* prefix). Confirmed with @dolauli that we cannot customize the behaviour. So I have to implement it myself.

[the remark this answers is not in the document]
Comment (BethanyZhou) [reply src/SignalR/SignalR.Autorest/custom/scripts/New-AzWebPubSubEventHubEndpointObject.ps1:28]: It surprised me. the generated cmdlets don't include prefix.

**Review by BethanyZhou — APPROVED** [review]

**Comment by Nickcandy** [comment]
@Y-Sindo Hi there, there are several mistakes when I migrate SignalR to Main branch. As you can see in log : https://dev.azure.com/azclitools/public/_build/results?buildId=16993&view=logs&j=b9afb22f-0837-5d02-bc24-6c30aee08bcf&t=990f6f37-4a21-5927-092a-220f38a460e9
Please correct these error as soon as possible so that we can release ur change.

**Pull request merged**